src/rustup/command.rs

```diff
@@ -30,8 +30,17 @@ pub fn run_command_for_dir<S: AsRef<OsStr>>(cmd: Command,
 
 fn telemetry_rustc<S: AsRef<OsStr>>(mut cmd: Command, args: &[S], cfg: &Cfg) -> Result<()> {
     let now = Instant::now();
-
+    
     cmd.args(&args[1..]);
+    
+    if stderr_isatty() && !(&args).iter().any(|e| {
+            let e = e.as_ref().to_str().unwrap_or("");
+            e == "--color"
+        }) 
+    {
```

Comment:
Please pull out this expression from `(&args).iter()` onward into a local variable with a descriptive name like, `has_color_arg`.

I believe this check should be `e.starts_with("--color")` since it could see e.g. `--color=always`.